python/pyarrow/table.pxi

```diff
@@ -344,6 +344,73 @@ cdef class ChunkedArray(_PandasConvertible):
 
         return pyarrow_wrap_chunked_array(result)
 
+    def filter(self, mask):
+        """
+        Filter the chunked array with a boolean mask.
+
+        Parameters
+        ----------
+        mask : Array or ChunkedArray
+            The boolean mask indicating which values to extract.
+
+        Returns
+        -------
+        ChunkedArray
+
+        Examples
+        --------
+        >>> import pyarrow as pa
+        >>> arr = pa.chunked_array([["a", "b"], ["c", None, "e"]])
+        >>> mask = pa.chunked_array([[True, False], [None, False, True]])
+
+        >>> arr.filter(mask)
+        <pyarrow.lib.ChunkedArray object at 0x7f8070081ea8>
+        [
+          [
+            "a"
+          ],
+          [
+            null,
+            "e"
+          ]
+        ]
+        """
+        # cdef:
+        #     cdef CDatum out
+
+        # with nogil:
+        #     check_status(
+        #         FilterKernel(_context(), CDatum(self.sp_chunked_array),
+        #                      CDatum(mask.sp_array), &out))
+
+        # return wrap_datum(out)
```

Comment:
Currently, the C++ Filter version that accepts Datums only accepts Array datums. If that would accept Datums of all kinds (and do the dispatch to the specialized Filter version there), the code here could be simpler (it could be something as the above). 

Would that be a desired feature of the C++ Filter kernel function?

Reply:
I brought this up when the ChunkedArray Filter functions were first implemented, the tracking JIRA is https://issues.apache.org/jira/browse/ARROW-7009

Reply:
So it's ok to leave that for a separate PR and go ahead here with the different overloads?